### PR TITLE
chore(Search, Typescript): Updates return type of `index.get` to `IndexWithPermissions` type.

### PR DESCRIPTION
**Pull request by jakeglobus**
- https://github.com/globusonline/globus-search/pull/1528/files

### CODE REVIEW

File: src/services/search/service/search-index.ts [hunk 1]
```diff
@@ -36,13 +36,16 @@ export type IndexList = {
...
+ * the client, but it is not a field in GSearchIndex itself.
+ *
  * @see https://docs.globus.org/api/search/reference/index_show/
  */
 export const get = function (
   index_id,
   options?,
   sdkOptions?,
-): Promise<JSONFetchResponse<GSearchIndex>> {
+): Promise<JSONFetchResponse<IndexWithPermissions>> {
```
Comment (jbottigliero) [inline src/services/search/service/search-index.ts:48]: I'm not seeing this reflected in the documentation yet, is it something that is still awaiting release?

Comment (jakeglobus) [reply src/services/search/service/search-index.ts:48]: It's been released, I believe the documentation has lagged behind. I'm totally fine waiting to merge this until the docs have been updated. In fact, I'll go ahead an put in a PR to update them

Comment (jakeglobus) [reply src/services/search/service/search-index.ts:48]: https://github.com/globusonline/docs.globus.org/pull/1759

**Review by jbottigliero — APPROVED** [review]

**Comment by jakeglobus** [comment]
Looks like I don't have authorization to merge, feel free to do so or add me to the authorized list. Thanks Joe

**Pull request merged**